lib/KTextTruncator.vue

```diff
@@ -0,0 +1,138 @@
+<template>
+
+  <!--
+    Text is wrapped in two `spans`s to allow parent components adding
+    padding style directly on `<TextTruncatorCss>` component no matter
+    of what truncating technique is used. Otherwise adding padding directly
+    would break when using technique (B) since text that should be truncated
+    would show in padding area.
+
+    Attributes are inherited by the inner `span` to emulate the same behavior
+    like if only one element would wrap the text to allow attributes be applied
+    as close as possible to the text element.
+
+    Some width information need to be provided to `<span>s` to allow `text-overflow`
+    calculate properly when ellipsis should be added.
+  -->
+  <span :style="{ display: 'inline-block', maxWidth: '100%' }">
+    <span
+      v-bind="$attrs"
+      :style="{ display: 'inline-block', maxWidth: '100%' }"
+      :class="$computedClass(truncate)"
+    >
+      {{ text }}
+    </span>
+  </span>
+
+</template>
+
+
+<script>
+
+  /**
+   * Truncates text to a certain number of lines
+   * and adds an ellipsis character "…"
+   *
+   * Internet Explorer note:
+   * Depending on length of words of the text, there might
+   * be a gap between the last visible word and "…"
+   */
+  export default {
+    name: 'KTextTruncator',
+    inheritAttrs: false,
+    props: {
+      text: {
+        type: String,
+        required: true,
+      },
+      maxLines: {
+        type: Number,
+        required: false,
+        default: 1,
+      },
+      /**
+       * Text line height in rem.
+       * Used only for Internet Explorer fallback.
+       */
+      lineHeightIE: {
+        type: Number,
+        required: false,
+        default: 1.4,
+      },
+    },
+    computed: {
+      truncate() {
+        /*
+          (A)
+          For one line, use standard ellipsis text overflow
+          that works well for such scenario
+        */
+        if (this.maxLines === 1) {
+          return {
+            overflow: 'hidden',
+            textOverflow: 'ellipsis',
+            whiteSpace: 'nowrap',
+          };
+        }
+
+        // 54 is random number only to be able to define `supports` test condition
+        if (process.client && 'CSS' in window && CSS.supports && CSS.supports('-webkit-line-clamp: 54')) {
```

Comment:
This is not necessary, just nice to have.

In other places in KDS, we usually define this constant `const nuxtServerSideRendering = process && process.server;` so that by its name it's clear immediately what the check is about. It could be nice to do something similar here.